rolecall/rolecall.py

```diff
@@ -101,73 +46,166 @@ class RoleCall:
+        keyring = self.settings[server_id][roleboard_id][message_id]
+        role_name = keyring[entry.emoji]['ROLE_NAME']
+        role = await self.get_or_create('role', role_name, entry.server)
+        return role
 
     @commands.group(pass_context=True, no_pm=True)
-    async def roleboard(self, ctx):
-        """change roleboard settings"""
+    async def rolecall(self, ctx):
+        """ Add emojis to a message where each emoji corresponds to a chosen role. 
+        If emoji is clicked, the corresponding role is assigned to the user. """
         server = ctx.message.server
 
         if ctx.invoked_subcommand is None:
             await self.bot.send_cmd_help(ctx)
         else:
             self.settings.setdefault(server.id, deepcopy(DEFAULT_SETTINGS))
 
-    @roleboard.command(pass_context=True, name="channel", no_pm=True)
-    async def roleboard_channel(self, ctx, channel: discord.Channel=None):
-        """Set the roleboard for this server.
+    @rolecall.command(pass_context=True, name="add", no_pm=True)
+    async def rolecall_add(self, ctx, channel: discord.Channel, 
+                            content_or_message_id: str, role: str, 
+                            emoji: str, 
+                            private_channel: str = None,
+                            ):
+        """
+        Add a role to a message. 
+        
+        channel
+
+           The channel where the new message will be posted or the channel
+           where the existing message is located
 
-        Leave blank to turn off the roleboard
+        content_or_message_id
+
+           Contents of the message. If the message
+           already exists, provide the message id instead.
+
+        role 
+
+           Name of the role. If a non-existing role is provided, 
+           it will be created for you.
+
+        emoji 
+
+           Emoji corresponding to the role which users will click
+           on.
+ 
+        private_channel(Optional) 
+
+           A private channel that members of the role will be granted access to. 
+           If a non-existing channel is provided, it will be created for you.
         """
         server = ctx.message.server
-        #channel = ctx.message.channel
         author = ctx.message.author
 
-        """
-        TODO: limit amount of messages. when over,
-            log warning and DM server owner /
-            infractor (if he's adding a new invite)
-        ^ invalid todo due to updated implementation notes
-        """
+        # retrieve role mentions in the command message
+        roles = ctx.message.raw_role_mentions
+        if len(roles) == 1:
+            role_object = discord.utils.get(server.roles, id=roles[0])
+        else:
+            role_object = await self.get_or_create("role", role, server)
 
-        settings = self.settings[server.id]
+        # retrieve channel mentions in the command message
+        channels = ctx.message.raw_channel_mentions
+        if len(channels) == 2:
```

Comment:
this and your raw_role_mentions has incorrect logic.
`[p]rolecall add chan1 "test @role1 #chan2 #chan3" role2 :eyes: chan4`
will cause the mentions in the message to be used instead of the role and channel in the actual argument

Reply:
applied correct logic as of 47f7037